pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/DeletePatchFactory.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.dsm.model.patch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.NotImplementedException;
+import org.broadinstitute.dsm.db.dao.ddp.tissue.TissueDao;
+import org.broadinstitute.dsm.db.dao.ddp.tissue.TissueSMIDDao;
+import org.broadinstitute.dsm.model.NameValue;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.NotificationUtil;
+
+public class DeletePatchFactory {
+
+    private DeletePatchFactory() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static BasePatch produce(Patch patch, NotificationUtil notificationUtil) {
+        BasePatch patcher;
+        if (DBConstants.DDP_ONC_HISTORY_DETAIL_ALIAS.equals(patch.getTableAlias())) {
+            patcher = new DeleteOncHistoryPatch(patch, notificationUtil);
+        } else if (PatchFactory.isTissueRelatedOncHistoryId(patch)) {
+            patcher = new DeleteTissuePatch(patch, notificationUtil);
+        } else {
+            throw new NotImplementedException("This method should not reach here");
+        }
+        return patcher;
+    }
+
+    protected static void setDeletedForChildrenFields(@NonNull Patch originalPatch, NotificationUtil notificationUtil) {
+        List<Patch> deletePatches = null;
+        if (originalPatch.getNameValue().getName().equals("t.deleted") && originalPatch.getTableAlias().equals(DBConstants.DDP_TISSUE_ALIAS)) {
+            deletePatches = getPatchForSmIds(originalPatch);
+        } else if (originalPatch.getNameValue().getName().equals("oD.deleted") &&
+                originalPatch.getTableAlias().equals(DBConstants.DDP_ONC_HISTORY_DETAIL_ALIAS)) {
+            deletePatches = getPatchForTissues(originalPatch);
+        }
+        for (Patch childPatch : deletePatches) {
+            BasePatch patcher = PatchFactory.makePatch(childPatch, notificationUtil);
+            patcher.doPatch();
+        }
```

Comment:
This method finds the tissues that belong to this onc history that just got deleted, or the sm ids that belong to the tissue,  and creates `Patch` requests for each of those tissues. The `doPatch()` method of those patches will then be called one by one